src/routes/(app)/cart/[id]/+page.server.ts

```diff
@@ -112,6 +112,33 @@ export const actions = {
 			depositPercentage
 		});
 
+		throw redirect(303, request.headers.get('referer') || '/cart');
+	},
+	addNote: async ({ locals, params, request }) => {
+		const product = await collections.products.findOne({ _id: params.id });
+		if (!product) {
+			await collections.carts.updateOne(userQuery(userIdentifier(locals)), {
+				$pull: { items: { productId: params.id } },
+				$set: { updatedAt: new Date() }
+			});
+			throw error(404, 'This product does not exist');
+		}
+		const formData = await request.formData();
+		const quantity = 1;
+		const { note } = z
+			.object({
+				note: z.string()
+			})
+			.parse({
+				note: formData.get('note')
+			});
+
+		await addToCartInDb(product, quantity, {
+			user: userIdentifier(locals),
+			totalQuantity: true,
+			note: note
+		});
```

Comment:
I don't think you should call `addToCartInDB`

Instead you can do:

```ts
const cart = await collections.cart.findOne(userQuery(userIdentifier(locals));

if (!cart) {
  throw ...;
}

const res = await collections.cart.updateOne({_id: cart._id, "items.productId": params.id}, {
	$set: {"$product.internalNote": ...}
});

if (!res.matchedCount) {
  throw ... // cart in DB doesn't have the product
}
```

You can remove all the product checking code.